tests/auth/test_environment_variables_are_loaded_correctly.py

```diff
@@ -0,0 +1,248 @@
+from typing import Optional
+
+import pydantic
+import pytest
+
+from django_setup_configuration.model_utils import create_config_source_models
+from django_setup_configuration.models import ConfigurationModel
+from testapp.models import SomeModel
+
+
+class MyConfigModel(ConfigurationModel):
+    nested_obj: Optional[dict] = pydantic.Field()
+
+    class Meta:
+        django_model_refs = {SomeModel: ("foo",)}
+
+
+class MyListedConfigModel(ConfigurationModel):
+    items: list[MyConfigModel]
+
+
+@pytest.fixture()
+def valid_configuration_obj_with_value_from_env():
+    return {
+        "extra": "Extra vars at the root level are valid",
+        "config_enabled": True,
+        "the_namespace": {
+            "foo": {
+                "value_from": {
+                    "env": "FOO",
+                }
+            },
+            "nested_obj": {"nested_foo": "a nested string"},
+        },
+    }
+
+
+@pytest.fixture()
+def yaml_file_with_valid_configuration_from_env(
+    yaml_file_factory, valid_configuration_obj_with_value_from_env
+):
+    yaml_path = yaml_file_factory(valid_configuration_obj_with_value_from_env)
+    return yaml_path
+
+
+@pytest.fixture()
+def yaml_file(request, yaml_file_factory):
+    marker = request.node.get_closest_marker("yaml_configuration")
+    yaml = {"config_enabled": True, "the_namespace": marker.args[0]}
+    return yaml_file_factory(yaml)
```

Comment:
Since we have this fixture, are the ones above needed?

Then we can specify the configuration above the test and the fixture makes a file of it?